src/components/data-exploration/embedding/Embedding.jsx

```diff
@@ -307,21 +315,25 @@ const Embedding = (props) => {
           ? (
             <ClusterPopover
               visible
-              popoverPosition={cellCoordintes}
+              popoverPosition={cellCoordinates}
               onCreate={onCreateCluster}
               onCancel={onCancelCreateCluster}
             />
           ) : (
             (cellInfoVisible && cellInfoTooltip.current) ? (
               <div>
                 <CellInfo
+                  width={CELLINFO_WIDTH}
                   componentType={embeddingType}
-                  coordinates={cellCoordintes}
+                  coordinates={cellCoordinates.current}
                   cellInfo={cellInfoTooltip.current}
+                  numTextRows={numCellInfoTextRows}
+                  invertX={cellCoordinates.current.x + CELLINFO_WIDTH > width}
+                  invertY={cellCoordinates.current.y + (numCellInfoTextRows * EM) + CELLINFO_Y_PADDING > height}
```

Comment:
I wouldn't calculate whether it needs to be inverted or not from the Embedding or from the Heatmap components -- this is something internal to the CellInfo component and if we do it outside of CellInfo, it only creates unnecessary coupling.

Reply:
you can pass width of the Embedding or the Heatmap as props to CellInfo and do those calculations inside CellInfo